docs/messaging/gcp-pubsub.md

```diff
@@ -16,7 +16,49 @@ For Google Cloud Pub/Sub, the following additional attributes are defined:
 <!-- semconv messaging.gcp_pubsub(full,tag=tech-specific-gcp-pubsub) -->
 | Attribute  | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
+| [`messaging.gcp_pubsub.message.link`](../attributes-registry/messaging.md) | int | An identifier for the link. This is *not* the message id. The number will be in between 0 and the max batch size. | `1` | Recommended |
 | [`messaging.gcp_pubsub.message.ordering_key`](../attributes-registry/messaging.md) | string | The ordering key for a given message. If the attribute is not present, the message does not have an ordering key. | `ordering_key` | Conditionally Required: If the message type has an ordering key set. |
 <!-- endsemconv -->
 
+## Examples
+
+### Asynchronous Batch Publish Example
+
+Given is a process P that asynchronously publishes 2 messages in a batch to a topic T on Pub/Sub.
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction LR
+  CA[Span Create A]
+  CB[Span Create B]
+  P[Span Publish]
+  end
+  CA-. link .-P;
+  CB-. link .-P;
+
+  classDef normal fill:green
+  class PA,PB,D1 normal
+  linkStyle 0,1 color:green,stroke:green
+```
```

Comment:
Thanks for adding this, that's awesome! I'd love to go over this in our next workgroup meeting if you can make it.

Reply:
No prob! I won't be able to make this week (US has Thanksgiving, but we can discuss next week!)

Reply:
Discussed today.